src/coreclr/src/gc/gc.cpp

```diff
@@ -16291,13 +16291,13 @@ void gc_heap::gc1()
             int limit = settings.condemned_generation;
             if (limit == max_generation)
             {
-                limit = total_generation_count;
+                limit = total_generation_count-1;
             }
             for (int gen = 0; gen < limit; gen++)
             {
                 size_t total_desired = 0;
 
-                for (int i = 0; i < gc_heap::n_heaps; i++)
+                for (int i = 0; i <= gc_heap::n_heaps; i++)
```

Comment:
"<=" should be in the outer for loop on limit, not here.

Reply:
Right! It was about generations, not heaps.